numpyro/distributions/copula.py

```diff
@@ -0,0 +1,108 @@
+    ]
+
+    def __init__(
+        self,
+        marginal_dist,
+        correlation_matrix=None,
+        correlation_cholesky=None,
+        *,
+        validate_args=None,
+    ):
+        if len(marginal_dist.event_shape) > 0:
+            raise ValueError("`marginal_dist` needs to be a univariate distribution.")
+
+        self.marginal_dist = marginal_dist
+        self.base_dist = MultivariateNormal(
+            covariance_matrix=correlation_matrix,
+            scale_tril=correlation_cholesky,
+        )
+
+        event_shape = self.base_dist.event_shape
+        batch_shape = lax.broadcast_shapes(
+            self.marginal_dist.batch_shape[:-1],
+            self.base_dist.batch_shape,
+        )
+
+        super(GaussianCopula, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+
+        shape = sample_shape + self.batch_shape
+        normal_samples = self.base_dist.expand(shape).sample(key)
+        cdf = Normal().cdf(normal_samples)
+        return self.marginal_dist.icdf(cdf)
+
+    @validate_sample
+    def log_prob(self, value):
+        # Ref: https://en.wikipedia.org/wiki/Copula_(probability_theory)#Gaussian_copula
+        # see also https://github.com/pyro-ppl/numpyro/pull/1506#discussion_r1037525015
+        marginal_lps = self.marginal_dist.log_prob(value)
+        quantiles = Normal().icdf(value)
+        #
+        copula_lp = (
+            self.base_dist.log_prob(quantiles)
+            + 0.5 * (quantiles**2).sum(-1)
+            + 0.5 * jnp.log(2 * jnp.pi) * quantiles.shape[-1]
+        )
+        return copula_lp + marginal_lps.sum(axis=-1)
+
+    @property
+    def mean(self):
+        return jnp.broadcast_to(self.marginal_dist.mean, self.shape())
+
+    @constraints.dependent_property(is_discrete=False, event_dim=1)
+    def support(self):
+        return constraints.independent(self.marginal_dist.support, 1)
+
+
+class GaussianCopulaBeta(GaussianCopula):
+    arg_constraints = {
+        "concentration1": constraints.positive,
+        "concentration0": constraints.positive,
+        "correlation_matrix": constraints.corr_matrix,
+        "correlation_cholesky": constraints.corr_cholesky,
+    }
+    support = constraints.independent(constraints.unit_interval, 1)
+
+    def __init__(
+        self,
+        concentration1,
+        concentration0,
+        correlation_matrix=None,
+        correlation_cholesky=None,
+        *,
+        validate_args=False,
+    ):
+        super().__init__(
+            Beta(concentration1, concentration0),
+            correlation_matrix,
+            correlation_cholesky,
+            validate_args=validate_args,
+        )
+        self.concentration1, self.concentration0 = promote_shapes(
+            concentration1, concentration0, shape=self.batch_shape + self.event_shape
+        )
```

Comment:
Could you add a jax unflatten method
```
    @classmethod
    def tree_unflatten(cls, aux_data, params):
        marginal_flatten, correlation_cholesky = params
        marginal_cls, marginal_aux = aux_data
        marginal_dist = marginal_cls.tree_unflatten(marginal_aux, marginal_flatten)
        return cls(marginal_dist.concentration1, marginal_dist.concentration0, correlation_cholesky=correlation_cholesky)
```

Reply:
@hessammehr we need to update the Beta version because
```
return cls(marginal_dist, correlation_cholesky=correlation_cholesky)
```
in the general class does not work here